hw8/task1.py

```diff
@@ -0,0 +1,149 @@
+            for line in f:
+                key, value = line.strip().split("=")
+                if value.isdigit():
+                    value = int(value)
+                super().__getattribute__("_file_storage")[key] = value
+
+    def __getattribute__(self, item: str):
+        """
+        Get attribute value through . notation
+        """
+        # try access to inner attributes and methods through . notation
+        try:
+            inner_data = super().__getattribute__(item)
+        except AttributeError:
+            ...  # do nothing
+        else:
+            return inner_data
+        # or obtain item from file storage
+        return super().__getattribute__("_file_storage").get(item)
+
+    def __getitem__(self, item: str) -> Any:
+        """
+        Get attribute value via square brackets
+        """
+        try:
+            res = super().__getattribute__("_file_storage").get(item)
+        except AttributeError:
+            self.__missing__(item)
+        return res
+
+    def __setattr__(self, name: str, value: Any):
+        """
+        Set attribute value through . notation
+        """
+        # a name presence and a value type check
+        file_path = super().__getattribute__("_file_path")
+        file_storage = super().__getattribute__("_file_storage")
+        if name not in file_storage:
+            raise AttributeError
+        if not isinstance(value, type(file_storage[name])):
+            raise ValueError
+
+        change_or_delete_line_from_file(file_path, name, value)
+        file_storage[name] = value
+
+    def __setitem__(self, key, value):
+        """
+        Set attribute value via square brackets
+        """
+        # a name presence and a value type check
+        file_path = super().__getattribute__("_file_path")
+        file_storage = super().__getattribute__("_file_storage")
+        if key not in file_storage:
+            raise AttributeError
+        if not isinstance(value, type(file_storage[key])):
+            raise ValueError
+
+        change_or_delete_line_from_file(file_path, key, value)
+        file_storage[key] = value
+
+    def __delitem__(self, key: str):
+        """
+        Delete item from _file_storage and file
+        """
+        file_path = super().__getattribute__("_file_path")
+        file_storage = super().__getattribute__("_file_storage")
+        if key not in file_storage:
+            raise AttributeError
+
+        change_or_delete_line_from_file(file_path, key, None, delete=True)
+        file_storage.pop(key)
+
+    def __getattr__(self, item: str):
+        """
+        Is called by __getattribute__ when AttributeError is risen
+        """
+        raise AttributeError
+
+    def __missing__(self, key: str):
+        """
+        Is called by __getitem__ when AttributeError is risen
+        """
+        print(f"There is no {key}")
+        raise AttributeError
+
+    def _get_copy_of_storage_dict(self) -> dict:
```

Comment:
not a good idea to mix test logic in the class

Reply:
Deleted it. Done